commit 4c8fe4f52755d4690a745f4e56b543c51add86fe
Author: Arnd Bergmann <arnd@arndb.de>
Date:   Fri Nov 20 15:24:39 2015 +0100

    regulator: core: avoid unused variable warning
    
    commit fa731ac7ea04a7d3a5c6d2f568132478c02a83b3 upstream.
    
    The second argument of the mutex_lock_nested() helper is only
    evaluated if CONFIG_DEBUG_LOCK_ALLOC is set. Otherwise we
    get this build warning for the new regulator_lock_supply
    function:
    
    drivers/regulator/core.c: In function 'regulator_lock_supply':
    drivers/regulator/core.c:142:6: warning: unused variable 'i' [-Wunused-variable]
    
    To avoid the warning, this restructures the code to make it
    both simpler and to move the 'i++' outside of the mutex_lock_nested
    call, where it is now always used and the variable is not
    flagged as unused.
    
    We had some discussion about changing mutex_lock_nested to an
    inline function, which would make the code do the right thing here,
    but in the end decided against it, in order to guarantee that
    mutex_lock_nested() does not introduced overhead without
    CONFIG_DEBUG_LOCK_ALLOC.
    
    Signed-off-by: Arnd Bergmann <arnd@arndb.de>
    Fixes: 9f01cd4a915 ("regulator: core: introduce function to lock regulators and its supplies")
    Link: http://permalink.gmane.org/gmane.linux.kernel/2068900
    Signed-off-by: Mark Brown <broonie@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/regulator/core.c b/drivers/regulator/core.c
index 73b7683..c70017d 100644
--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -138,18 +138,10 @@ static bool have_full_constraints(void)
  */
 static void regulator_lock_supply(struct regulator_dev *rdev)
 {
-	struct regulator *supply;
-	int i = 0;
-
-	while (1) {
-		mutex_lock_nested(&rdev->mutex, i++);
-		supply = rdev->supply;
-
-		if (!rdev->supply)
-			return;
+	int i;
 
-		rdev = supply->rdev;
-	}
+	for (i = 0; rdev->supply; rdev = rdev->supply->rdev, i++)
+		mutex_lock_nested(&rdev->mutex, i);
 }
 
 /**
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1240_4c8fe4f52755d4690a745f4e56b543c51add86fe.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1240_4c8fe4f52755d4690a745f4e56b543c51add86fe.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
